docs/madr/decisions/056-policy-matching-with-real-resources.md

```diff
@@ -0,0 +1,185 @@
+# Policy matching algorithm with real resources (MeshService, MeshExternalService, Mesh*Route)
+
+* Status: accepted
+
+## Context and Problem Statement
+
+We can think of the policy matching algorithm as a function
+that maps a set of policies provided by the user to the following structure:
+
+```yaml
+Rules:
+  - Subset:
+      - Key: kuma.io/service
+        Not: false
+        Value: foo
+    Conf:
+      connectionTimeout: 2s
+      http:
+        requestTimeout: 15s
+      idleTimeout: 20s
+    Origin: [ ... ]
+  - Subset:
+      - Key: kuma.io/service
+        Not: false
+        Value: bar
+    Conf:
+      connectionTimeout: 2s
+      http:
+        requestTimeout: 20s
+      idleTimeout: 20s
+    Origin: [ ... ]
+```
+
+This structure is used by both "from" and "to" policies plugins.
+Policy plugins use the `Rules` list to compute `Conf` for the known destination.
+In example, if we want to know the `Conf` for outbound with `kuma.io/service: bar` tags,
+we're going to iterate over `Rules` and check if our outbound belongs to the `Rules[].Subset`.
+
+The approach has a number of disadvantages:
+
+* even though it works well for "from" policies, using it for "to" policies doesn't feel natural
```

Comment:
Can you elaborate? I don't share the same feeling - maybe I'm missing some context.